Prefabs/Locomotion/BodyRepresentation/SharedResources/Scripts/BodyRepresentationInternalSetup.cs

```diff
@@ -277,13 +289,87 @@ public virtual void Process()
             EmitIsGroundedChangedEvent(isGrounded);
         }
 
+        /// <summary>
+        /// Solves body collisions by not moving the body in case it can't go to its current position.
+        /// </summary>
+        /// <remarks>
+        /// If body collisions should be prevented this method needs to be called right before or right after applying any form of movement to the body.
+        /// </remarks>
+        public virtual void SolveBodyCollisions()

```

Comment:
This was kinda on the public facade on purpose - see the remarks section to understand why a user may want to have direct access to this.

One could argue that it's more internal and the user just has to understand that once they have a reason to call this themselves they are forced to work with these internals in a way after all.

Reply:
I'm dumb. It's still available on the facade, just proxied into this InternalSetup!